package.json

```diff
@@ -50,7 +56,7 @@
   },
   "dependencies": {
     "atom-languageclient": "0.9.7",
-    "javascript-typescript-langserver": "^2.7.0"
+    "typescript-language-server": "0.3.6"
```

Comment:
v0.3.6 doesn't have the hierarchical support fixes merged (the latest few 0.4.0-dev versions have it). Additionally none of these versions have the child_process forking implemented, its only implemented in PRs right now. Without the forking changes the shebang at the top of `tsserver` will be used which requires users have node installed locally and that PATH happens to resolve correctly (which it doesn't on osx using homebrew to install node).

Reply:
I'd rather we move on without waiting for 0.4.0 for now as we actually already have a hierarchy anyway.

I did try using your fork but the yarn dependency was making things just plain not install for myself on Windows and another team member on macOS.  I think we're better off either putting something in the README or some detection logic about node being required on the path until they merge the logic in upstream.

Reply:
I think it might be best to deploy a built fork of the package to npm then with the forking logic in. Note some very interesting hacks I needed to get it working without: https://github.com/mattlyons0/ide-typescript-theia/commit/f218f1dcaffcca692151fa0e7a048b0bedd11a64#diff-16b7d75b70953cbf4755508170f2f24cL34
The node binary wasn't resolving on startup which caused the plugin to crash, after ~10sec when startup had completed then I could disable and enable the plugin to restart everything and it would resolve. I put this hack in to get it to resolve on startup by having the user specify the folder which `node` binary was in.

I've built a version of `typescript-language-server` from https://github.com/mattlyons0/typescript-language-server/tree/release and published it to npm under `@mattlyons/typescript-language-server`. Feel free to give that a shot and grab the (tarball)[https://github.com/mattlyons0/typescript-language-server/releases/download/v0.4.0/mattlyons-typescript-language-server-0.4.0.tgz] if you wish to publish it under a package you own.